### PR TITLE
refactor(localdebug): fix eslint error in localdebug

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Detail: 
[Task 24659964](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24659964): Fix eslint error in vscode-extension package debug
[Task 24659838](https://msazure.visualstudio.com/Microsoft%20Teams%20Extensibility/_workitems/edit/24659838): eslint promise error for xiaofu

### CODE REVIEW

File: packages/vscode-extension/src/debug/depsChecker/vscodeLogger.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ export class VSCodeLogger implements DepsLogger {
   }
 
   private addToDetailCache(level: LogLevel, message: string): void {
-    const line = `${LogLevel[level]} ${new Date().toISOString()}: ${message}`;
+    const line = `${String(LogLevel[level])} ${new Date().toISOString()}: ${message}`;
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/depsChecker/vscodeLogger.ts:59]: Keep consistency? `String()` or `as string` for all changes

File: packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts [hunk 2]
```diff
@@ -41,13 +40,17 @@ export abstract class BaseTaskTerminal implements vscode.Pseudoterminal {
     this.do()
       .then((res) => {
         const error = res.isErr() ? res.error : undefined;
-        this.stop(error);
+        this.stop(error).catch((error) => {
+          this.writeEmitter.fire(`${error?.message as string}\r\n`);
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts:44]: Is this newly added logic?

File: packages/vscode-extension/src/debug/depsChecker/vscodeLogger.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/depsChecker/vscodeLogger.ts:59]: Now I am using `String()` to convert other types (such as numbers, booleans, and undefined values) into strings and using the type assertion `as string` to convert a variable of type any to a string.

File: packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts (hunk 2 above)
Comment (XiaofuHuang) [reply packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts:44]: Yes, we need to add `catch` to fix eslint problem. And add a log to show the error happened in `stop`.

Comment (swatDong) [reply packages/vscode-extension/src/debug/taskTerminal/baseTaskTerminal.ts:44]: I see. So this is also a bug fix for the missing error handling on `stop`.

**Review by swatDong — APPROVED** [review]

**Review by nliu-ms — APPROVED** [review]

**Review by xzf0587 — APPROVED** [review]

**Pull request merged**